deploy/bin/add-k3s-cluster-to-config.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+# Copy the cluster information defined in the k3s config file to the user's kubeconfig file
+# If the user already has a kubeconfig file, add the cluster information to the existing file
+# otherwise create a new kubeconfig file
+
+set -ex
+
+# Don't use `k3s kubectl` here because it always uses the default kubeconfig file in /etc/rancher/k3s/k3s.yaml
+K=${KUBECTL_CMD:-kubectl}
+
+k3scfg="/etc/rancher/k3s/k3s.yaml"
+
+kubectl_is_k3s=$($K version --client | egrep "^Client Version.*+k3s" || true)
+
+if [ -n "$kubectl_is_k3s" ]; then
+    echo "Using the kubectl supplied by k3s. Shared kubeconfig file is at ${k3scfg}"
+    echo "If you want to use a different kubeconfig, set the KUBECONFIG environment variable"
+    exit 0
+fi
+
+if [ -f $k3scfg ]; then
+    # Get the cluster and user information from the k3s config file
+    server=$($K config view --kubeconfig="${k3scfg}" --minify --flatten --output 'jsonpath={.clusters[?(@.name=="default")].cluster.server}')
+    cert=$($K config view --kubeconfig="${k3scfg}" --minify --flatten --output 'jsonpath={.clusters[?(@.name=="default")].cluster.certificate-authority-data}')
+
+    user_client_certificate=$(kubectl config view --kubeconfig=${k3scfg} --minify --flatten --output 'jsonpath={.users[?(@.name=="default")].user.client-certificate-data}')
+    user_client_key=$(kubectl config view --kubeconfig=${k3scfg} --minify --flatten --output 'jsonpath={.users[?(@.name=="default")].user.client-key-data}')
+
+    # Add the cluster information to the user's kubeconfig file
```

Comment:
Maybe we could throw a warning or an error if there already exists a k3s context in the kubeconfig file?

Reply:
Probably shouldn't be an error and I'm not convinced it should be a warning. Generally, I think that means that we have uninstalled k3s and are reinstalling it or some variation of that. So all the cluster and user get changed and the old ones aren't valid anymore.

I suppose we could check to see if someone had a context called `k3s` that was pointing at something other than the local k3s instance, but do we think that's a real thing?